agent/internal/client/base_client.go

```diff
@@ -1025,3 +1025,20 @@ func (c *baseCommunicator) AssumeRole(ctx context.Context, td TaskData, request
 	}
 	return &creds, nil
 }
+
+func (c *baseCommunicator) AllowedSingleTaskDistroTasks(ctx context.Context) ([]string, error) {
```

Comment:
Looks like this method + the route aren't needed anymore since the logic is now in the next task route.